compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1975,13 +1975,51 @@ class Typer extends Namer
           val exprPt = pt.baseType(defn.QuotedExprClass)
           val quotedPt = if (exprPt.exists) exprPt.argTypesHi.head else defn.AnyType
           val quoted1 = typedExpr(quoted, quotedPt)(quoteContext.addMode(Mode.QuotedPattern))
-          val (shape, splices) = splitQuotePattern(quoted1)
-          val patType = defn.tupleType(splices.tpes.map(_.widen))
-          val splicePat = typed(untpd.Tuple(splices.map(untpd.TypedSplice(_))).withSpan(quoted.span), patType)
+
+          val (typeBindings, shape, splices) = splitQuotePattern(quoted1)
```

Comment:
Add a description of the scheme for translating patterns to doc comment above

Reply:
Added